docs/src/tutorials/getting_started/design_patterns_for_larger_models.jl

```diff
@@ -0,0 +1,610 @@
+
+function _add_knapsack_variables(
+    model::JuMP.Model,
+    data::_KnapsackData,
+    ::BinaryKnapsack,
+)
+    return JuMP.@variable(model, x[keys(data.objects)], Bin)
+end
+
+function _add_knapsack_variables(
+    model::JuMP.Model,
+    data::_KnapsackData,
+    ::IntegerKnapsack,
+)
+    return JuMP.@variable(model, x[keys(data.objects)] >= 0, Int)
+end
+
+function _add_knapsack_constraints(
+    model::JuMP.Model,
+    data::_KnapsackData,
+    ::_AbstractKnapsack,
+)
+    x = model[:x]
+    JuMP.@constraint(
+        model,
+        capacity_constraint,
+        sum(v.weight * x[k] for (k, v) in data.objects) <= data.capacity,
+    )
+    return
+end
+
+function _add_knapsack_objective(
+    model::JuMP.Model,
+    data::_KnapsackData,
+    ::_AbstractKnapsack,
+)
+    x = model[:x]
+    JuMP.@objective(model, Max, sum(v.profit * x[k] for (k, v) in data.objects))
+    return
+end
+
+function _solve_knapsack(
+    optimizer,
+    data::_KnapsackData,
+    config::_AbstractKnapsack,
+)
+    model = JuMP.Model(optimizer)
+    _add_knapsack_variables(model, data, config)
+    _add_knapsack_constraints(model, data, config)
+    _add_knapsack_objective(model, data, config)
+    JuMP.optimize!(model)
+    if JuMP.termination_status(model) != JuMP.OPTIMAL
+        @warn("Model not solved to optimality")
+        return nothing
+    end
+    return JuMP.value.(model[:x])
+end
+
+"""
+    solve_knapsack(optimizer, data_filename::String, config::_AbstractKnapsack)
+
+Solve the knapsack problem and return the optimal primal solution
+
+## Arguments
+
+ * `optimizer` : an object that can be passed to `JuMP.Model` to construct a new
+   JuMP model.
+ * `data_filename` : the filename of a JSON file containing the data for the
+   problem.
+ * `config` : an object to control the type of knapsack model constructed.
+   Valid options are:
+    * `BinaryKnapsack()`
+    * `IntegerKnapsack()`
+
+## Returns
+
+ * If an optimal solution exists: a `JuMP.DenseAxisArray` that maps the `String`
+   name of each object to the number of objects to pack into the knapsack.
+ * Otherwise, `nothing`, indicating that the problem does not have an optimal
+   solution.
+
+## Examples
+
+```julia
+solution = solve_knapsack(GLPK.Optimizer, "path/to/data.json", BinaryKnapsack())
+```
+
+```julia
+solution = solve_knapsack(
+    MOI.OptimizerWithAttributes(GLPK.Optimizer, "msg_lev" => 0),
+    "path/to/data.json",
+    IntegerKnapsack(),
+)
+```
+"""
+function solve_knapsack(
+    optimizer,
+    data_filename::String,
+    config::_AbstractKnapsack,
+)
+    return _solve_knapsack(optimizer, _read_data(data_filename), config)
+end
+
+end
+
+# Finally, you can call your model:
+
+import .KnapsackModel
+
+KnapsackModel.solve_knapsack(
+    GLPK.Optimizer,
+    joinpath(@__DIR__, "data", "knapsack.json"),
+    KnapsackModel.BinaryKnapsack(),
+)
+
+# !!! note
+#     The `.` in `.KnapsackModel` denotes that it is a submodule and not a
```

Comment:
Great tip. Did not know this and other's likely won't either.

Reply:
The wording here is slightly cheating. It means "Load `@__module__.KnapsackModel`, but I wasn't sure how to express this without getting too technical. 

Note that you can do things like `import ..Foo` as well for relative imports. We use that in MOI a fair bit.